src/Umbraco.Web/Strategies/RelateOnTrashHandler.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Linq;
+using Umbraco.Core;
+using Umbraco.Core.Auditing;
+using Umbraco.Core.Events;
+using Umbraco.Core.Models;
+using Umbraco.Core.Services;
+
+namespace Umbraco.Web.Strategies
+{
+    public sealed class RelateOnTrashHandler : ApplicationEventHandler
+    {
+        protected override void ApplicationStarting(UmbracoApplicationBase umbracoApplication, ApplicationContext applicationContext)
```

Comment:
Good spot!
Yeah, I'd pretty much copied the `RelateOnCopyHandler` class as the basis.